src/modules/auth/passport/oidc-strategy.ts

```diff
@@ -39,16 +46,47 @@ passport.use(
       );
     }
 
+    if (
+        !process.env.OIDC_PROVIDER ||
+        !process.env.OIDC_CONFIG ||
+        !process.env.OIDC_CLIENT_ID ||
+        !process.env.OIDC_CLIENT_SECRET ||
+        !process.env.OIDC_REDIRECT_URI
+    ) {
+        logger.error('Not all OIDC environment variables are properly defined');
+        callback('Internal server error');
```

Comment:
The custom `HttpApiException` already includes a `statusCode` field which is used by passportjs to show sensible errors on the frontend. If the `statusCode` is not passed, the default is `500`. For consistency and clarity reasons, I suggest to use the `HttpApiException`.

Furthermore, note that after calling `callback(err)`, the function should return. If not, while the correct status code is returned, the rest of the function is still tried to be executed, which can lead to unexpected behaviour.

```suggestion
        callback(new HttpApiException(HttpStatusCode.InternalServerError));
        return
```